firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistenceTest.java

```diff
@@ -0,0 +1,261 @@
+        .setSdkVersion("sdkVersion")
+        .setGmpAppId("gmpAppId")
+        .setPlatform(1)
+        .setInstallationUuid("installationId")
+        .setBuildVersion("1")
+        .setDisplayVersion("1.0.0")
+        .setSession(makeTestSession(sessionId))
+        .build();
+  }
+
+  private static CrashlyticsReport.Session makeTestSession(String sessionId) {
+    return Session.builder()
+        .setGenerator("generator")
+        .setIdentifier(sessionId)
+        .setStartedAt(0)
+        .setApp(makeTestApplication())
+        .build();
+  }
+
+  private static Application makeTestApplication() {
+    return Application.builder()
+        .setIdentifier("applicationId")
+        .setVersion("version")
+        .setDisplayVersion("displayVersion")
+        .build();
+  }
+
+  private static Event makeTestEvent() {
+    return Event.builder()
+        .setType("type")
+        .setTimestamp(1000)
+        .setApp(
+            Session.Event.Application.builder()
+                .setBackground(false)
+                .setExecution(
+                    Execution.builder()
+                        .setBinaries(
+                            ImmutableList.from(
+                                Execution.BinaryImage.builder()
+                                    .setBaseAddress(0)
+                                    .setName("name")
+                                    .setSize(100000)
+                                    .setUuid("uuid")
+                                    .build()))
+                        .setException(
+                            Execution.Exception.builder()
+                                .setFrames(makeTestFrames())
+                                .setOverflowCount(0)
+                                .setReason("reason")
+                                .setType("java.lang.Exception")
+                                .build())
+                        .setSignal(Signal.builder().setCode("0").setName("0").setAddress(0).build())
+                        .setThreads(
+                            ImmutableList.from(
+                                Session.Event.Application.Execution.Thread.builder()
+                                    .setName("name")
+                                    .setImportance(4)
+                                    .setFrames(makeTestFrames())
+                                    .build()))
+                        .build())
+                .setUiOrientation(1)
+                .build())
+        .setDevice(
+            Session.Event.Device.builder()
+                .setBatteryLevel(0.5)
+                .setBatteryVelocity(3)
+                .setDiskUsed(10000000)
+                .setOrientation(1)
+                .setProximityOn(true)
+                .setRamUsed(10000000)
+                .build())
+        .build();
+  }
+
+  private static ImmutableList<Frame> makeTestFrames() {
+    return ImmutableList.from(
+        Frame.builder()
+            .setPc(0)
```

Comment:
It's probably a better test if these aren't all 0.

Reply:
Good call, I'll update this.